element_array_ephys/spike_sorting/ecephys_spike_sorting.py

```diff
@@ -0,0 +1,326 @@
+        ).fetch1("acq_software", "clustering_method")
+        assert clustering_method in (
+            "kilosort2",
+            "kilosort2.5",
+            "kilosort3",
+        ), 'Supporting "kilosort" clustering_method only'
+
+        params = (KilosortPreProcessing & key).fetch1("params")
+
+        if acq_software == "SpikeGLX":
+            spikeglx_meta_filepath = ephys.get_spikeglx_meta_filepath(key)
+            spikeglx_recording = spikeglx.SpikeGLX(spikeglx_meta_filepath.parent)
+            spikeglx_recording.validate_file("ap")
+
+            run_kilosort = kilosort_triggering.SGLXKilosortPipeline(
+                npx_input_dir=spikeglx_meta_filepath.parent,
+                ks_output_dir=kilosort_dir,
+                params=params,
+                KS2ver=f'{Decimal(clustering_method.replace("kilosort", "")):.1f}',
+                run_CatGT=True,
+            )
+            run_kilosort._modules = ["kilosort_helper"]
+            run_kilosort._CatGT_finished = True
+            run_kilosort.run_modules()
+        elif acq_software == "Open Ephys":
+            oe_probe = ephys.get_openephys_probe_data(key)
+
+            assert len(oe_probe.recording_info["recording_files"]) == 1
+
+            # run kilosort
+            run_kilosort = kilosort_triggering.OpenEphysKilosortPipeline(
+                npx_input_dir=oe_probe.recording_info["recording_files"][0],
+                ks_output_dir=kilosort_dir,
+                params=params,
+                KS2ver=f'{Decimal(clustering_method.replace("kilosort", "")):.1f}',
+            )
+            run_kilosort._modules = ["kilosort_helper"]
+            run_kilosort.run_modules()
+
+        self.insert1(
+            {
+                **key,
+                "execution_time": execution_time,
+                "execution_duration": (
+                    datetime.utcnow() - execution_time
+                ).total_seconds()
+                / 3600,
+            }
+        )
+
+
+@schema
+class KilosortPostProcessing(dj.Imported):
+    """A processing table to handle each clustering task."""
+
+    definition = """
+    -> KilosortClustering
+    ---
+    modules_status: longblob   # dictionary of summary status for all modules
+    execution_time: datetime   # datetime of the start of this step
+    execution_duration: float  # (hour) execution duration
+    """
+
+    def make(self, key):
+        execution_time = datetime.utcnow()
+
+        output_dir = (ephys.ClusteringTask & key).fetch1("clustering_output_dir")
+        kilosort_dir = find_full_path(ephys.get_ephys_root_data_dir(), output_dir)
+
+        acq_software, clustering_method = (
+            ephys.ClusteringTask * ephys.EphysRecording * ephys.ClusteringParamSet & key
+        ).fetch1("acq_software", "clustering_method")
+        assert clustering_method in (
+            "kilosort2",
+            "kilosort2.5",
+            "kilosort3",
```

Comment:
Given that this tuple is reused, should it migrate elsewhere as a SUPPORTED_KILOSORT_VERSIONS constant?